File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/wallet/CreateWalletTest.java
package org.hyperledger.indy.sdk.wallet;

import static org.hamcrest.CoreMatchers.isA;

import java.util.concurrent.ExecutionException;

import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.json.JSONObject;
import org.junit.Test;


public class CreateWalletTest extends IndyIntegrationTest {

	@Test
	public void testCreateWalletWorks() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testCreateWalletWorksForEmptyType() throws Exception {
		String config = new JSONObject()
				.put("id", WALLET)
				.toString();

		Wallet.createWallet(config, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testCreateWalletWorksForUnknowType() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(UnknownWalletTypeException.class));

		String config =
				new JSONObject()
						.put("id", WALLET)
						.put("storage_type", "unknown_type")
						.toString();

		Wallet.createWallet(config, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testCreateWalletWorksForEmptyName() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		String config = new JSONObject()
				.put("id", "")
				.toString();
		Wallet.createWallet(config, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testCreateWalletWorksForDuplicate() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletExistsException.class));

		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/CreateWalletTest.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class CreateWalletTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestCreateWalletWorks()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
        }
        
        [TestMethod]
        public async Task TestCreateWalletWorksForEmptyType()
        {
            var config = JsonConvert.SerializeObject(new { id = WalletUtils.GetWalletId() });
            await Wallet.CreateWalletAsync(config, WALLET_CREDENTIALS);
        }


        [TestMethod]
        public async Task TestCreateWalletWorksForUnknownType()
        {
            var config = JsonConvert.SerializeObject(new {
                id = WalletUtils.GetWalletId(),
                storage_type = "unknown_type"
            });

            var ex = await Assert.ThrowsExceptionAsync<UnknownWalletTypeException>(() =>
                Wallet.CreateWalletAsync(config, WALLET_CREDENTIALS)
            );
        }

        [TestMethod]
        public async Task TestCreateWalletWorksForEmptyName()
        {
            var config = JsonConvert.SerializeObject(new { id = string.Empty });

            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                Wallet.CreateWalletAsync(config, WALLET_CREDENTIALS)
            );
        }

        [TestMethod]
        public async Task TestCreateWalletFailsForDuplicate()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var ex = await Assert.ThrowsExceptionAsync<WalletExistsException>(() =>
               Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS)
            );
        }
    }
}
